### ISSUE TITLE
module 'weasyprint' has no attribute 'fonts'

### ISSUE THREAD

**Issue by hnb2** [fdemmer/django-weasyprint#53]
Hi there,

Weasyprint released a version 53 of their library yesterday (release notes here: https://github.com/Kozea/WeasyPrint/releases/tag/v53.0) and it changed an import:

> FontConfiguration is now in the weasyprint.text.fonts module.

Because setup.py uses a `>=` instead of a `==`, when we rebuilt our app yesterday everything stopped working because of the import.

Example in your `make test`:

```
======================================================================
ERROR: test_get_pdf (django_weasyprint.tests.test_views.WeasyTemplateViewTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pierre/Dev/django-weasyprint/django_weasyprint/tests/test_views.py", line 17, in test_get_pdf
    response = self.client.get('/pdf/')
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/test/client.py", line 535, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/test/client.py", line 345, in get
    return self.generic('GET', path, secure=secure, **{
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/test/client.py", line 422, in generic
    return self.request(**r)
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/test/client.py", line 503, in request
    raise exc_value
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/core/handlers/base.py", line 145, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/core/handlers/base.py", line 143, in _get_response
    response = response.render()
  File "/home/pierre/Dev/django-weasyprint/.tox/py38-dj22/lib/python3.8/site-packages/django/template/response.py", line 106, in render
    self.content = self.rendered_content
  File "/home/pierre/Dev/django-weasyprint/django_weasyprint/views.py", line 94, in rendered_content
    document = self.get_document()
  File "/home/pierre/Dev/django-weasyprint/django_weasyprint/views.py", line 77, in get_document
    font_config = self.get_font_config()
  File "/home/pierre/Dev/django-weasyprint/django_weasyprint/views.py", line 52, in get_font_config
    return weasyprint.fonts.FontConfiguration()
AttributeError: module 'weasyprint' has no attribute 'fonts'
```

I created a PR which fixes this problem here: https://github.com/fdemmer/django-weasyprint/pull/52

Thank you

**Comment by n1ngu** [comment]
Precisely due to weasyprint [not following semver](https://doc.courtbouillon.org/weasyprint/latest/api_reference.html#versioning) (see https://github.com/Kozea/WeasyPrint/issues/900 ) maybe it would be worth to republish the django-weasyprint 1.X series as they were but requiring "weasyprint>=43,<=52"

I'd suggest to at least republish
- 1.1.0 --> 1.1.0.post1
- 1.0.2 --> 1.0.2.post1

so as to ease dependency resolution for any future installation https://www.python.org/dev/peps/pep-0440/#post-releases

**Comment by ThomSawyer** [comment]
WeasyPrint version 52.5 seems to work as well without this erro message. Maybe we could set the restriction to `weasyprint>=43,<53`.

**Comment by SchrodingersGat** [comment]
Just ran into this one too. Any chance that weasyprint 53 will be supported soon?

**Comment by fdemmer** [comment]
Thank you to everybody helping out here!

I just released [.post1](https://pypi.org/project/django-weasyprint/1.1.0.post1/) versions as @n1ngu suggested pinning WeasyPrint<53 as @ThomSawyer suggested :)

As for the suggested compatibility code: I'll not merge this and continue to try and keep up with WeasyPrint's non-semver releases, because not only that import path changed, but generating formats other than PDF was removed completely in 53.

Therefore I'll make a version 2.0 requiring WeasyPrint >=53 soon.
